posts/TEMPLATE/index.md

```diff
@@ -1,9 +1,117 @@
 ---
 date: 2030-12-31 # Auto updates on publish.
-title: Hello World!
-author: sascha.lampalzer@cybertec.at
-tags: ["list-tags-here", "🙂"] # max. 10 tags; lowercase; dash-separated
-description: "This is an example" # max. 300 chars.
+title: Quickly test whether ports can be reached on your linux machine.
+author: julian.markwort@cybertec.at
+tags: ["firewall", "linux", "netcat", "nc", "ports", "testing", "probe", "listening address"] # max. 10 tags; lowercase; dash-separated
+description: "How would you quickly check if a given port is reachable on your system or not, so you can go complain to the network admins? I'll show you my way." # max. 300 chars.
 ---
 
-This is just an example post, don't worry about it. We use this internally as a template for awesome, future posts.
+If you've ever tried to launch anything that opens a port on your linux machine and you'd like to know whether you've misconfigured something (e.g. the listen address), or whether you're running into firewall issues, you'll be grateful to find out that there's a simple way to test for firewall issues.
+
+There's a tool called `netcat`; It's part of almost any linux server setup I've worked on and it can be launched by typing `nc` on your command line.
+netcat has two modes:
+- a calling mode
+- a listening mode
+
+We can use both of these, on two sides of our server to check if the firewall will let us pass or not.
+
+(insert thou shall not pass image)
+
+## Introduction to netcat
+
+On the machine where you'd like to run the server, which will be listen on a given port, say 54321, run this to open a listening netcat:
+```
+nc -vlk 54321
+```
+I'll break those arguments down for you:
+- `v` is for verbose: print more details about connections
+- `l` is for listening mode
+- `k` is for keep-open; This means that netcat will accept connections over and over again.
+- 54321 is for the listening port. By default, netcat listens on all IP-addresses it can find, but you can also limit it to a given IP-address, say your primary IP-address, so your service can be reached from your network, but not via localhost, for example.
```

Comment:
```suggestion
- `54321` is for the listening port. By default, `netcat` listens on all IP-addresses it can find, but you can also limit it to a given IP-address, say your primary IP-address, so your service can be reached from your network, but not via localhost, for example.
```